build.gradle.kts

```diff
@@ -29,4 +35,37 @@ subprojects {
 
 tasks.register("clean", Delete::class) {
     delete(rootProject.buildDir)
-}
\ No newline at end of file
+}
+
+tasks.withType<DependencyUpdatesTask> {
+    rejectVersionIf {
+        isNonStable(candidate.version)
+    }
+    gradleReleaseChannel = "current"
+}
+
+tasks.named<DependencyUpdatesTask>("dependencyUpdates").configure {
+    outputFormatter = "html"
+}
+
+versionCatalogUpdate {
+    versionCatalogs {
+        create("appLibs") {
+            catalogFile.set(file("gradle/dependencies.toml"))
+        }
+        create("testLibs") {
+            catalogFile.set(file("gradle/testDependencies.toml"))
+        }
+
+        create("androidTestLibs") {
+            catalogFile.set(file("gradle/androidTestDependencies.toml"))
+        }
+    }
+}
+
+fun isNonStable(version: String): Boolean {
+    val stableKeyword = listOf("RELEASE", "FINAL", "GA").any { version.toUpperCase().contains(it) }
+    val regex = "^[0-9,.v-]+(-r)?$".toRegex()
```

Comment:
I think it'd be nice to have a short comment breaking down/explaining the regex so whoever uses this is aware of what's checking

Reply:
I've documented the function. Let me know if it makes sense